homeassistant/components/knx/config_flow.py

```diff
@@ -0,0 +1,254 @@
+        )
+
+    async def async_step_tunnel(self, user_input: dict | None = None) -> FlowResult:
+        """Select a tunnel from a list. Will be skipped if the gateway scan was unsuccessful or if only one gateway was found."""
+        errors: dict = {}
+
+        if user_input is not None:
+            gateway: GatewayDescriptor = next(
+                filter(
+                    lambda _gateway: str(_gateway)
+                    == user_input[CONF_KNX_GATEWAY],  # type: ignore
+                    self._tunnels,
+                )
+            )
+
+            self._gateway_ip = gateway.ip_addr
+            self._gateway_port = gateway.port
+
+            return await self.async_step_manual_tunnel()
+
+        tunnel_repr = {
+            str(tunnel) for tunnel in self._tunnels if tunnel.supports_tunnelling
+        }
+
+        #  skip this step if the user has only one unique gateway.
+        if len(tunnel_repr) == 1:
+            _gateway: GatewayDescriptor = self._tunnels[0]
+            self._gateway_ip = _gateway.ip_addr
+            self._gateway_port = _gateway.port
+            return await self.async_step_manual_tunnel()
+
+        fields = {vol.Required(CONF_KNX_GATEWAY): vol.In(tunnel_repr)}
+
+        return self.async_show_form(
+            step_id="tunnel", data_schema=vol.Schema(fields), errors=errors
+        )
+
+    async def async_step_routing(self, user_input: dict | None = None) -> FlowResult:
+        """Routing setup."""
+        errors: dict = {}
+
+        if user_input is not None:
+            return self.async_create_entry(
+                title=CONF_KNX_ROUTING,
+                data={
+                    ConnectionSchema.CONF_KNX_MCAST_GRP: user_input[
+                        ConnectionSchema.CONF_KNX_MCAST_GRP
+                    ],
+                    ConnectionSchema.CONF_KNX_MCAST_PORT: user_input[
+                        ConnectionSchema.CONF_KNX_MCAST_PORT
+                    ],
+                    CONF_KNX_INDIVIDUAL_ADDRESS: user_input[
+                        CONF_KNX_INDIVIDUAL_ADDRESS
+                    ],
+                    CONF_KNX_CONNECTION_TYPE: CONF_KNX_ROUTING,
+                },
+            )
+
+        fields = OrderedDict()
+        fields[
+            vol.Required(CONF_KNX_INDIVIDUAL_ADDRESS, default=XKNX.DEFAULT_ADDRESS)
+        ] = str
+        fields[
+            vol.Required(ConnectionSchema.CONF_KNX_MCAST_GRP, default=DEFAULT_MCAST_GRP)
+        ] = str
+        fields[
+            vol.Required(
+                ConnectionSchema.CONF_KNX_MCAST_PORT, default=DEFAULT_MCAST_PORT
+            )
+        ] = cv.port
+
+        return self.async_show_form(
+            step_id="routing", data_schema=vol.Schema(fields), errors=errors
+        )
+
+    async def async_step_import(self, config: dict | None = None) -> FlowResult:
+        """Import a config entry.
+
+        Special type of import, we're not actually going to store any data.
```

Comment:
This docstring doesn't match the code. We *are* creating an entry here and storing data.